pageserver/src/tenant_mgr.rs

```diff
@@ -321,12 +310,88 @@ pub async fn detach_tenant(
     // we will attempt to remove files which no longer exist. This can be fixed by having shutdown
     // mechanism for tenant that will clean temporary data to avoid any references to ephemeral files
     let local_tenant_directory = conf.tenant_path(&tenant_id);
-    fs::remove_dir_all(&local_tenant_directory).with_context(|| {
-        format!(
-            "Failed to remove local tenant directory '{}'",
-            local_tenant_directory.display()
-        )
-    })?;
+    tokio::fs::remove_dir_all(&local_tenant_directory)
+        .await
+        .with_context(|| {
+            format!("Failed to remove local tenant directory {local_tenant_directory:?}")
+        })?;
+
+    Ok(())
+}
+
+pub async fn schedule_tenant_load(
+    conf: &'static PageServerConf,
+    tenant_id: TenantId,
+    remote_storage: Option<GenericRemoteStorage>,
+) -> anyhow::Result<()> {
+    let tenant_path = conf.tenant_path(&tenant_id);
+    anyhow::ensure!(
+        tenant_path.is_dir(),
+        "Cannot load tenant from path {tenant_path:?}, it either does not exist or not a directory"
+    );
+    anyhow::ensure!(
+        !crate::is_temporary(&tenant_path),
+        "Cannot load tenant from temporary path {tenant_path:?}"
+    );
+    anyhow::ensure!(
+        !tenant_path.is_empty_dir().with_context(|| {
+            format!("Failed to check whether {tenant_path:?} is an empty dir")
+        })?,
+        "Cannot load tenant from empty directory {tenant_path:?}"
+    );
+    let tenant_attach_mark = conf.tenant_attaching_mark_file_path(&tenant_id);
+    anyhow::ensure!(
+        !tenant_attach_mark.exists(),
+        "Tenant attach is in progress: attach mark is at {tenant_attach_mark:?}"
+    );
+
+    run_if_no_tenant_in_memory(tenant_id, |vacant_entry| {
+        let tenant_ignore_mark = conf.tenant_ignore_mark_file_path(tenant_id);
+        if tenant_ignore_mark.exists() {
+            std::fs::remove_file(&tenant_ignore_mark)
+                .with_context(|| format!("Failed to remove tenant ignore mark {tenant_ignore_mark:?} during tenant loading"))?;
+        }
+
+        let new_tenant = load_local_tenant(conf, &tenant_path, remote_storage)
+            .with_context(|| format!("Failed to load tenant from path {tenant_path:?}"))?;
```

Comment:
Error message should make it clear that we only initiated the load/attach operation.
Admittedly, the `load_local_tenant` function name is bad, since it can spawn either a load or an attach operation internally.

Maybe rename to `spawn_tenant`? Or `start_tenant`? That's generic over load vs. attach

Reply:
Names are somewhat odd indeed.
I've named it as `schedule_local_tenant_processing` which does not shine, but at least have some verb, just "spawning" or "starting" does not sound ok to me.